xds/src/main/java/io/grpc/xds/internal/sds/XdsServerBuilder.java

```diff
@@ -199,26 +201,30 @@ public static XdsServerBuilder forPort(int port) {
 
   @Override
   public Server build() {
+    XdsClientWrapperForServerSds xdsClient = new XdsClientWrapperForServerSds(port);
+    ServerSdsProtocolNegotiator serverProtocolNegotiator = null;
     if (fallbackProtocolNegotiator != null) {
-      ServerSdsProtocolNegotiator serverProtocolNegotiator =
-          SdsProtocolNegotiators.serverProtocolNegotiator(port, fallbackProtocolNegotiator);
-      return buildServer(serverProtocolNegotiator);
-    } else {
-      return new ServerWrapperForXds(
-          delegate.build(), new XdsClientWrapperForServerSds(port), errorNotifier);
+      serverProtocolNegotiator =
+          SdsProtocolNegotiators.serverProtocolNegotiator(fallbackProtocolNegotiator);
     }
+    return buildServer(xdsClient, serverProtocolNegotiator);
   }
 
   /**
    * Creates a Server using the given serverSdsProtocolNegotiator: gets the
```

Comment:
Remove the 2nd sentence in the comment since it doesn't "get the getXdsClientWrapperForServerSds from the serverSdsProtocolNegotiator". In any case with my PR this will be a non-public method so the whole comment can be removed.

Reply:
Done. Also added a mention of xdsClient.